crates/prql-ast/src/error.rs

```diff
@@ -0,0 +1,107 @@
+//! Error message produced by the compiler.
+// Should be in some prql_common crate, but until we need that, it can reside here.
```

Comment:
FYI I thought this blog post was interesting, which I read yesterday — I haven't really thought about it like this before, given I've never worked on a _huge_ rust project. 

https://blog.danhhz.com/compile-times-and-code-graphs